### PR TITLE
Add GeneralizedHarmonic Outflow BC

### PR DESCRIPTION
**Pull request by moxcodes**
## Proposed changes

Adds the outflow boundary correction, which simply checks that all characteristic speeds are directed out of the domain, to the generalized harmonic system.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/Outflow.hpp [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+  using dg_interior_evolved_variables_tags = tmpl::list<>;
+  using dg_interior_temporary_tags = tmpl::list<
+      ::GeneralizedHarmonic::ConstraintDamping::Tags::ConstraintGamma1,
+      gr::Tags::Lapse<DataVector>,
+      gr::Tags::Shift<Dim, Frame::Inertial, DataVector>>;
+  using dg_gridless_tags = tmpl::list<>;
+  using dg_interior_primitive_variables_tags = tmpl::list<>;
+
+  static std::optional<std::string> dg_outflow(
```
Comment (moxcodes) [inline src/Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/Outflow.hpp:65]: Move to cpp

File: src/Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/Outflow.hpp [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+#include "Options/Options.hpp"
+#include "Parallel/CharmPupable.hpp"
+#include "PointwiseFunctions/GeneralRelativity/SpacetimeNormalVector.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace GeneralizedHarmonic::BoundaryConditions {
+template <size_t Dim>
+class Outflow final : public BoundaryCondition<Dim> {
```
Comment (nilsdeppe) [inline src/Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/Outflow.hpp:30]: some Doxygen would be nice so it shows up in the documentation :)

File: src/Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/Outflow.hpp [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+#include <memory>
+#include <optional>
+#include <pup.h>
+#include <string>
+#include <type_traits>
+
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tags/TempTensor.hpp"
+#include "DataStructures/Tensor/EagerMath/DeterminantAndInverse.hpp"
```
Comment (nilsdeppe) [inline src/Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/Outflow.hpp]: you don't need several of the includes here (might be worth double checking in the GRMHD PR now that things got moved to the cpp)

File: src/Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/Outflow.cpp [hunk 4]
```diff
@@ -0,0 +1,78 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/Outflow.hpp"
+
+#include <cstddef>
+#include <memory>
+#include <pup.h>
+
+#include "Utilities/GenerateInstantiations.hpp"
```
Comment (nilsdeppe) [inline src/Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/Outflow.cpp:18]: you definitely need a lot more includes here now

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
you can squash immediately

**Comment by moxcodes** [comment]
fixes squashed in.
Cheers!

**Review by nilsdeppe — APPROVED** [review]

**Review by geoffrey4444 — APPROVED** [review]

**Comment by geoffrey4444** [comment]
Code looks good, and I have also successfully used this BC in some BBH evolutions

**Pull request merged**